example/lib/widgets/button.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/material.dart';
+
+import '../utils/dimensions.dart';
+
+class Button extends StatelessWidget {
+  const Button({
+    Key? key,
+    required this.onPressedFunction,
+    required this.enabled,
+    required this.text,
+    required this.style,
+    this.icon,
+  }) : super(key: key);
+  final void Function() onPressedFunction;
+  final bool enabled;
+  final String text;
+  final ButtonStyle style;
+  final IconData? icon;
+
+  @override
+  Widget build(BuildContext context) {
+    return icon == null
+        ? ElevatedButton(
+            style: style,
+            onPressed: enabled
+                ? () {
+                    onPressedFunction();
+                  }
+                : null,
```

Comment:
why this onPressed is different from the below version with "icon != null" ?

Reply:
Void removed